src/librustc/infer/region_inference/graphviz.rs

```diff
@@ -124,7 +125,7 @@ pub fn maybe_print_constraints_for<'a, 'gcx, 'tcx>(
 struct ConstraintGraph<'a, 'gcx: 'a+'tcx, 'tcx: 'a> {
     graph_name: String,
     region_rels: &'a RegionRelations<'a, 'gcx, 'tcx>,
-    map: &'a FxHashMap<Constraint<'tcx>, SubregionOrigin<'tcx>>,
+    map: &'a BTreeMap<Constraint<'tcx>, SubregionOrigin<'tcx>>,
```

Comment:
Nit: I would appreciate a comment here like this:

```
// Using `BTreeMap` because the order in which we iterate over these constraints
// can affect the way we build the region graph, which in turn affects the way
//  that region errors are reported, leading to small variations
// in error output across runs and platforms.
```

Reply:
Oh, actually, not *here*, but down below =)